enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,428 @@
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: control-plane
+      failureDomain: us-east-1a
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.241/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: control-plane
+      failureDomain: us-east-1b
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.242/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: control-plane
+      failureDomain: us-east-1c
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.243/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.244/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.245/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.246/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+~~~
+
+### Scaling in Additional Machines
+
+Additional machines requiring static IP addresses may be scaled in by:
+- An OpenShift administrator prepares a machine yaml with the static IP address included in the provider specification
+- An external controller assigns an IP address in response to an `IPAddressClaim` being created by the machine controller
+
+#### Machine yaml with embedded static IP address
+
+An administrator may create a bespoke machine configuration yaml which embeds static IP configuration. This allows a new machine to be provisioned without the need of an external controller.
+
+Example:
+~~~yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: Machine
+metadata:
+  name: test-compute-1
+spec:
+  metadata: {}
+  providerSpec:
+    value:
+      ...
+      network:
+        devices:
+          - networkName: lab
+            ipAddrs:
+            - 192.168.101.244/24
+            gateway4: 192.168.101.1
+            nameservers:
+            - 192.168.101.2            
+      ...
+~~~
+
+#### External controller assigns address
+
+An external controller may be used to automatically assign an IP address to a node during the provisioning process. This is accomplished by the definition of an IP address pool in the machineset/machine provider spec.
+
+Example:
+~~~yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  name: static-machineset-worker
+  namespace: openshift-machine-api
+spec:
+  replicas: 2
+  ...
+  template:
+    ...
+    spec:          
+      ...
+      network:
+        devices:
+        - networkName: port-group-vlan-101          
+          addressesFromPools:                   
+          - apiGroup: machine.openshift.io
+            kind: IPPool
+            name: example-pool
```

Comment:
I guess this came from upstream? It's shouldn't be kind, it should be resource 😞 When we do the API review we should discuss this

Reply:
Yes, this came from upstream.  We can pick up the discussion in the API review.  

https://github.com/openshift/enhancements/pull/1267#discussion_r1185982570

One of the benefits of keeping this is that the external IPAM controller isn't required during install.  We could eliminate the ability to define the static IPs in the spec and require the installation to be done on a compact cluster.  At present, even during an installation, control plane nodes don't embed static IP configuration.